src/main/java/org/logstash/dissect/fields/NormalField.java

```diff
@@ -0,0 +1,52 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public final class NormalField extends AbstractField {
+
+    private static final Field missing = new NormalField("missing field", 100000);
+    public static Field getMissing() {
+        return missing;
+    }
+
+    public static Field create(String s) {
+        return new NormalField(s);
+    }
+
+    private NormalField(String s) {
+        super(s, 1);
+    }
+
+    private NormalField(String s, int ord) {
+        super(s, ord);
+    }
+
+    @Override
+    public boolean saveable() {
+        return true;
+    }
+
+    @Override
+    public void append(Map<String, Object> keyValueMap, ValueResolver values) {
+        keyValueMap.put(this.name, values.get(this));
+    }
+
+    @Override
+    public void append(Event event, ValueResolver values) {
+      event.setField(this.name, values.get(this));
```

Comment:
Whether to overwrite a field in the Event or keyValueMap or not - is an addition to the config definition - I mean if users complain about overwritten fields then we can add a way of defining which fields are (or not) overwrite-able, however in general it is accepted in LS that successive plugin operations on an event may overwrite fields from before. Its kind of a progressive enhancement.
Dissect and Grok are most likely going to be the first plugin to handle an event.